File: soutLink-backend/src/main/java/com/soutLink/supervisor/dto/DetailedProjectResponse.java
package com.soutLink.supervisor.dto;

import com.soutLink.auth.UserDetailsResponse;
import com.soutLink.enums.ProjectStatus;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class DetailedProjectResponse {
    private Long id;
    private String title;
    private String description;
    private int numberStudents;
    private ProjectStatus status;
    private SupervisorResponse supervisor;
    private List<UserDetailsResponse> students;
    private List<FileResponse> files;
    private String createdDate;
}

File: soutLink-frontend/src/app/services/models/detailed-project-response.ts
/* tslint:disable */
/* eslint-disable */
import { FileResponse } from '../models/file-response';
import { SupervisorResponse } from '../models/supervisor-response';
import { UserDetailsResponse } from '../models/user-details-response';
export interface DetailedProjectResponse {
  createdDate?: string;
  description?: string;
  files?: Array<FileResponse>;
  id?: number;
  numberStudents?: number;
  status?: 'NOT_TAKEN' | 'IN_PROGRESS' | 'ACCEPTED';
  students?: Array<UserDetailsResponse>;
  supervisor?: SupervisorResponse;
  title?: string;
}
